datafusion/core/src/physical_plan/sort_merge_join.rs

```diff
@@ -1414,7 +1407,7 @@ mod tests {
             "| 3  | 5  | 9  | 20 | 5  | 80 |",
             "+----+----+----+----+----+----+",
         ];
-        assert_batches_sorted_eq!(expected, &batches);
+        assert_batches_eq!(expected, &batches);
```

Comment:
I think it might be worth a note here explaining that the sort order of the output is important in case someone might try and change it in the future:

```suggestion
        // The output order is important as SMJ preserves sortedness
        assert_batches_eq!(expected, &batches);
```